sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClient.java

```diff
@@ -188,31 +191,35 @@ public void uploadFromFile(String filePath, boolean overwrite) {
     public void uploadFromFile(String filePath, ParallelTransferOptions parallelTransferOptions,
         BlobHttpHeaders headers, Map<String, String> metadata, AccessTier tier, BlobRequestConditions requestConditions,
         Duration timeout) throws UncheckedIOException {
-        this.uploadFromFile(filePath, new BlobUploadFromFileOptions()
+        this.uploadFromFileWithResponse(filePath, new BlobUploadFromFileOptions()
                 .setParallelTransferOptions(parallelTransferOptions).setHeaders(headers).setMetadata(metadata)
                 .setTier(tier).setRequestConditions(requestConditions),
-            timeout);
+            timeout, null);
     }
 
     /**
      * Creates a new block blob, or updates the content of an existing block blob.
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * {@codesnippet com.azure.storage.blob.specialized.cryptography.EncryptedBlobClient.uploadFromFile#String-BlobUploadFromFileOptions-Duration}
+     * {@codesnippet com.azure.storage.blob.specialized.cryptography.EncryptedBlobClient.uploadFromFileWithResponse#String-BlobUploadFromFileOptions-Duration-Context}
      *
      * @param filePath Path of the file to upload
      * @param options {@link BlobUploadFromFileOptions}
      * @param timeout An optional timeout value beyond which a {@link RuntimeException} will be raised.
      * @throws UncheckedIOException If an I/O error occurs
+     * @return Information about the uploaded block blob.
      */
     @Override
-    public void uploadFromFile(String filePath, BlobUploadFromFileOptions options, Duration timeout)
+    public Response<BlockBlobItem> uploadFromFileWithResponse(String filePath, BlobUploadFromFileOptions options,
```

Comment:
Shouldnt filePath be part of the constructor of Options?

Reply:
Good point. I forgot about that part.

Reply:
Done.